types/vnode.d.ts

```diff
@@ -1,7 +1,8 @@
 import { Vue } from "./vue";
 
+export type ScopedSlot = (props: any) => VNode | VNode[] | string | undefined;
```

Comment:
I guess the return type should have similar type with `VNodeChildren` as it is normalized as same as vnode children.
https://github.com/vuejs/vue/blob/8a800867fe61e5aa642e1e3da91bb890d07312f7/src/core/vdom/helpers/normalize-scoped-slots.js#L42-L51

Maybe it will be:

```ts
type ScopedSlot = (props: any) => ScopedSlotReturn;
type ScopedSlotReturn = VNode | string | boolean | null | undefined | ScopedSlotReturnArray;
interface ScopedSlotReturnArray extends Array<ScopedSlotReturn> {}
```

Reply:
@KaelWD are you planning to update the PR according to the suggestion?